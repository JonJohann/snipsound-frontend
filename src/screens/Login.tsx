import React, { useEffect, useState } from 'react';
import { KeyboardAvoidingView, View, Text, TextInput, TouchableOpacity, Image, Platform, Keyboard } from 'react-native';
import { useDispatch } from 'react-redux';
import SnipsoundLogo from '../components/misc/SnipsoundLogo';
import { resetParams } from '../state/ParamsSlice';
import { loginClicked } from '../state/SoundSlice'
import { styles } from '../styles/styles'

// Screen for loging in to Snipsound
export default function Login({ navigation }: any) {

    const [username, setUsername] = useState("")
    const dispatch = useDispatch()
    const validUsername = () => username.length < 15 && username.length > 1

    const login = () => {
        dispatch(resetParams())
        dispatch(loginClicked(username));
        Keyboard.dismiss();
        navigation.navigate('SoundList');
    }
    
    return (
        <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
            <SnipsoundLogo />
            <KeyboardAvoidingView behavior={Platform.OS == "ios" ? "padding" : "height"} style={{}}>
                <Text style={[styles.title, { alignSelf: "center" }]}>What's your name?</Text>
                <View style={{ flexDirection: "row", justifyContent: "center" }}>
                    <TextInput onChangeText={text => setUsername(text)} style={[styles.textInput, { flex: 0, width: 200 }]} placeholder="Name.."></TextInput>
                    <TouchableOpacity onPress={login} style={[( validUsername() ? styles.active : styles.inactive), { marginTop: 10 }]}
                        disabled={!validUsername()}>
                        <Text style={{ fontSize: 20 , fontFamily: 'Manrope_400Regular' }}>Enter</Text>
                    </TouchableOpacity>
                </View>
                {username.length >= 15 ? <Text style={{color: "red", alignSelf: "center"}}>Name must be less than 15 characters</Text>: null}
            </KeyboardAvoidingView>
        </View>
    )
}