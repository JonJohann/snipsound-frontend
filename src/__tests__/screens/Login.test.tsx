import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import Login from '../../screens/Login';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('<Login />', () => {
    const initialState = { output: 10 }
    const mockStore = configureStore()
    let store = mockStore(initialState)

    beforeEach(() => {
        jest.useFakeTimers()
    })
    it('has 2 children', () => {
        const tree = renderer.create(<Provider store={store}><Login /></Provider>).toJSON() as ReactTestRendererJSON;
        expect(tree!.children!.length).toBe(2);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Provider store={store}><Login /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
      });
      
});