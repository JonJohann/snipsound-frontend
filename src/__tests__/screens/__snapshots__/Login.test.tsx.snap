// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "alignSelf": "stretch",
        "marginHorizontal": 10,
        "marginVertical": -110,
        "width": undefined,
      }
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {},
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Manrope_400Regular",
            "fontSize": 30,
            "fontWeight": "bold",
          },
          Object {
            "alignSelf": "center",
          },
        ]
      }
    >
      What's your name?
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Name.."
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomWidth": 0.5,
              "borderColor": "gray",
              "borderRadius": 10,
              "fontSize": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "paddingHorizontal": 10,
            },
            Object {
              "flex": 0,
              "width": 200,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#e6e6e6",
            "borderColor": "#6d557000",
            "borderRadius": 10,
            "borderWidth": 1,
            "elevation": 2,
            "marginHorizontal": 5,
            "marginTop": 10,
            "marginVertical": 5,
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Manrope_400Regular",
              "fontSize": 20,
            }
          }
        >
          Enter
        </Text>
      </View>
    </View>
  </View>
</View>
`;
